Code/Tools/ProjectManager/Source/ProjectExportWorker.cpp

```diff
@@ -0,0 +1,184 @@
+ *
+ */
+
+#include <ProjectExportWorker.h>
+#include <ProjectManagerDefs.h>
+#include <PythonBindingsInterface.h>
+#include <ProjectUtils.h>
+
+#include <QDir>
+#include <QFile>
+#include <QProcess>
+#include <QProcessEnvironment>
+#include <QTextStream>
+#include <QThread>
+
+namespace O3DE::ProjectManager
+{
+    ProjectExportWorker::ProjectExportWorker(const ProjectInfo& projectInfo)
+        : QObject()
+        , m_projectInfo(projectInfo)
+    {
+    }
+
+    void ProjectExportWorker::ExportProject()
+    {
+        auto result = ExportProjectForPlatform();
+
+        if (result.IsSuccess())
+        {
+            emit Done();
+        }
+        else
+        {
+            emit Done(result.GetError());
+        }
+    }
+
+    QString ProjectExportWorker::GetLogFilePath() const
+    {
+        QDir logFilePath(m_projectInfo.m_path);
+        // Make directories if they aren't on disk
+        if (!logFilePath.cd(ProjectBuildPathPostfix))
+        {
+            logFilePath.mkpath(ProjectBuildPathPostfix);
+            logFilePath.cd(ProjectBuildPathPostfix);
+
+        }
+        if (!logFilePath.cd(ProjectBuildPathCmakeFiles))
+        {
+            logFilePath.mkpath(ProjectBuildPathCmakeFiles);
+            logFilePath.cd(ProjectBuildPathCmakeFiles);
+        }
+        return logFilePath.filePath(ProjectExportErrorLogName);
+    }
+
+    void ProjectExportWorker::QStringToAZTracePrint([[maybe_unused]] const QString& error)
+    {
+        AZ_TracePrintf("Project Manager", error.toStdString().c_str());
+    }
+
+    //TODO: update this code to just run the desired worker process with the project path argument
+    AZ::Outcome<void, QString> ProjectExportWorker::ExportProjectForPlatform()
+    {
+        // Check if we are trying to cancel task
+        if (QThread::currentThread()->isInterruptionRequested())
+        {
+            QStringToAZTracePrint(ExportCancelled);
+            return AZ::Failure(ExportCancelled);
+        }
+
+        QFile logFile(GetLogFilePath());
+        if (!logFile.open(QIODevice::WriteOnly | QIODevice::Text | QIODevice::Truncate))
+        {
+            QString error = tr("Failed to open log file.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        EngineInfo engineInfo;
+
+        AZ::Outcome<EngineInfo> engineInfoResult = PythonBindingsInterface::Get()->GetEngineInfo();
+        if (engineInfoResult.IsSuccess())
+        {
+            engineInfo = engineInfoResult.GetValue();
+        }
+        else
+        {
+            QString error = tr("Failed to get engine info.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        QTextStream logStream(&logFile);
+        if (QThread::currentThread()->isInterruptionRequested())
+        {
+            logFile.close();
+            QStringToAZTracePrint(ExportCancelled);
+            return AZ::Failure(ExportCancelled);
+        }
```

Comment:
I don't know if you need to check for cancellation here and in the beginning of this function. I don't think its likely (though minutely possible) that they can hit cancel in between the time a log file is created and the engine info is set. If you decide to keep the cancel check here, you should also consider deleting the `logFile` as well.

Reply:
Upon further review I agree. The time between the threads is too small for any reasonable cancellation request (as only the Cancel button via user click can make the cancellation request directly). I will remove these checks